api/src/org/labkey/api/query/QuerySettings.java

```
@@ -226,31 +232,40 @@ public void init(PropertyValues pvs)
             {
                 setViewName(viewName);
             }
-            if (_getParameter(param(QueryParam.ignoreFilter)) != null)
+            String ignoreFilter = _getParameter(param(QueryParam.ignoreFilter));
+            if (isNotBlank(ignoreFilter))
+                _ignoreUserFilter = (Boolean) ConvertUtils.convert(ignoreFilter, Boolean.class);
+
+            String reportId = _getParameter(param(QueryParam.reportId));
+            if (isNotBlank(reportId))
             {
-                _ignoreUserFilter = true;
+                var identifier = ReportService.get().getReportIdentifier(reportId, null, null);
+                if (null == identifier)
+                    throw new NotFoundException();
+                setReportId(identifier);
             }
-
-            setReportId(ReportService.get().getReportIdentifier(_getParameter(param(QueryParam.reportId)), null, null));
         }
 
         // Ignore maxRows and offset parameters when not PAGINATED.
         if (_showRows == ShowRows.PAGINATED)
         {
             String offsetParam = _getParameter(param(QueryParam.offset));
-            if (offsetParam != null)
+            if (isNotBlank(offsetParam))
             {
                 try
                 {
                     long offset = Long.parseLong(offsetParam);
                     if (offset > 0)
                         _offset = offset;
                 }
-                catch (NumberFormatException ignored) { }
+                catch (NumberFormatException nfe)
+                {
+                    throw new BadRequestException(SC_BAD_REQUEST, "Could not parse parameter 'offset'", nfe);

```

Same thing here and a few other spots - include the bogus value in the error message